src/jit/codegenxarch.cpp

```diff
@@ -7478,8 +7504,8 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* putArgStk)
     // a separate putarg_stk for each of the upper and lower halves.
     noway_assert(targetType != TYP_LONG);
 
-    int argSize = genTypeSize(genActualType(targetType));
-    genStackLevel += argSize;
+    const unsigned argSize = putArgStk->getArgSize();
+    assert((argSize % TARGET_POINTER_SIZE) == 0);
 
     // TODO-Cleanup: Handle this in emitInsMov() in emitXArch.cpp?
```

Comment:
I would say not; I'll remove it.

---
In reply to: [88321469](https://github.com/dotnet/coreclr/pull/7847#discussion_r88321469) [](ancestors = 88321469)